engine/runtime/action-processor/src/geometry/convex_hull_constructor.rs

```diff
@@ -0,0 +1,209 @@
+use reearth_flow_geometry::types::geometry_collection::GeometryCollection;
+use reearth_flow_runtime::node::REJECTED_PORT;
+use reearth_flow_runtime::{
+    channels::ProcessorChannelForwarder,
+    errors::BoxedError,
+    event::EventHub,
+    executor_operation::{ExecutorContext, NodeContext},
+    node::{Port, Processor, ProcessorFactory, DEFAULT_PORT},
+};
+use reearth_flow_types::{Attribute, AttributeValue, Feature, GeometryValue};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use super::errors::GeometryProcessorError;
+
+#[derive(Debug, Clone, Default)]
+pub struct ConvexHullConstructorFactory;
+
+impl ProcessorFactory for ConvexHullConstructorFactory {
+    fn name(&self) -> &str {
+        "ConvexHullConstructor"
+    }
+
+    fn description(&self) -> &str {
+        "Constructs a convex hull"
+    }
+
+    fn parameter_schema(&self) -> Option<schemars::schema::RootSchema> {
+        Some(schemars::schema_for!(ConvexHullConstructorParam))
+    }
+
+    fn categories(&self) -> &[&'static str] {
+        &["Geometry"]
+    }
+
+    fn get_input_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn get_output_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone(), REJECTED_PORT.clone()]
+    }
+
+    fn build(
+        &self,
+        _ctx: NodeContext,
+        _event_hub: EventHub,
+        _action: String,
+        with: Option<HashMap<String, Value>>,
+    ) -> Result<Box<dyn Processor>, BoxedError> {
+        let param: ConvexHullConstructorParam = if let Some(with) = with {
+            let value: Value = serde_json::to_value(with).map_err(|e| {
+                GeometryProcessorError::ConvexHullConstructorFactory(format!(
+                    "Failed to serialize 'with' parameter: {}",
+                    e
+                ))
+            })?;
+            serde_json::from_value(value).map_err(|e| {
+                GeometryProcessorError::ConvexHullConstructorFactory(format!(
+                    "Failed to deserialize 'with' parameter: {}",
+                    e
+                ))
+            })?
+        } else {
+            return Err(GeometryProcessorError::ConvexHullConstructorFactory(
+                "Missing required parameter `with`".to_string(),
+            )
+            .into());
+        };
+        let process = ConvexHullConstructor {
+            group_by: param.group_by,
+            buffer_2d: HashMap::new(),
+        };
+
+        Ok(Box::new(process))
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ConvexHullConstructorParam {
+    group_by: Option<Vec<Attribute>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ConvexHullConstructor {
+    group_by: Option<Vec<Attribute>>,
+    buffer_2d: HashMap<String, Vec<Feature>>,
+}
+
+fn get_attributes(feature: &Feature, attributes: &[Attribute]) -> Vec<AttributeValue> {
```

Comment:
Please add it here.
https://github.com/reearth/reearth-flow/blob/main/engine/runtime/types/src/feature.rs#L259